nautobot/dcim/factory.py

```diff
@@ -99,6 +99,112 @@ def get_rack_reservation_units(obj):
     return [unit for unit in available_units if unit not in unavailable_units][:1]
 
 
+class DeviceFactory(PrimaryModelFactory):
+    class Meta:
+        model = Device
+        exclude = (
+            "has_tenant",
+            "has_platform",
+            "has_serial",
+            "has_asset_tag",
+            "has_device_redundancy_group",
+            "has_comments",
+        )
+
+    device_type = random_instance(DeviceType, allow_null=False)
+    status = random_instance(
+        lambda: Status.objects.get_for_model(Device),
+        allow_null=False,
+    )
+    role = random_instance(
+        lambda: Role.objects.get_for_model(Device),
+        allow_null=False,
+    )
+    location = random_instance(
+        lambda: Location.objects.filter(location_type__content_types=ContentType.objects.get_for_model(Device)),
+        allow_null=False,
+    )
+    name = factory.LazyAttributeSequence(lambda o, n: f"{o.device_type.model} Device - {o.location} - {n + 1}")
+
+    has_tenant = NautobotBoolIterator()
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+    has_platform = NautobotBoolIterator()
+    platform = factory.Maybe("has_platfrom", random_instance(Platform))
```

Comment:
```suggestion
    platform = factory.Maybe("has_platform", random_instance(Platform))
```

The Platform manufacturer should match the DeviceType manufacturer